Acrolinx.Sidebar.SDK/FileUtil.cpp

```diff
@@ -0,0 +1,164 @@
+#include "stdafx.h"
+#include "FileUtil.h"
+#include <string>
+#include "DllUtil.h"
+#include "Resource.h"
+
+EXTERN_C IMAGE_DOS_HEADER __ImageBase;
+
+Acrolinx_Sdk_Sidebar_Util::FileUtil::FileUtil(void)
+{
+}
+
+Acrolinx_Sdk_Sidebar_Util::FileUtil::~FileUtil(void)
+{
+}
+
+CString Acrolinx_Sdk_Sidebar_Util::FileUtil::ExtractEmbeddedStartPage()
+{
+    HINSTANCE hwnd = DllUtil::GetDllHandle();
+    HRSRC hResource = FindResource(hwnd, MAKEINTRESOURCE(IDR_STARTPAGE), L"BINARY");
+    HGLOBAL hResData = LoadResource(hwnd, hResource);
+    DWORD  resSize = SizeofResource(hwnd, hResource);
+
+    LPBYTE lpRes = (LPBYTE)LockResource(hResData);
+
+    if (!hResource || !hResData || !lpRes || !resSize) {
+        LOGE << "Error reading embedded start page resource";
+    }
+
+    WCHAR wcharPath[MAX_PATH];
+    CString startPageTempPath = _T("");
+    if (GetTempPathW(MAX_PATH, wcharPath))
+    {
+        startPageTempPath = wcharPath;
+    }
+
+    startPageTempPath = startPageTempPath + _T("Acrolinx\\");
+    CreateDirectory(startPageTempPath, NULL);
+
+    CString startPageZip = startPageTempPath + _T("startpage.zip");
+
+    startPageTempPath = startPageTempPath + _T("startpage-") + DllUtil::GetAppName() + _T("\\");
+    CreateDirectory(startPageTempPath, NULL);
+
+    HANDLE hFile = CreateFile(startPageZip, GENERIC_WRITE, 0, NULL, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, NULL);
+    if (hFile == INVALID_HANDLE_VALUE) {
+        // TODO: Handle error
+    }
+
+    DWORD dwBytesWritten = 0;
+    BOOL bSuccess = WriteFile(hFile, lpRes, resSize, &dwBytesWritten, NULL);
+    if (!bSuccess || dwBytesWritten != resSize) {
+        // TODO: Handle error
+    }
+
+    CloseHandle(hFile);
+
+    bool success = FileUtil::UnzipToFolder(startPageZip.AllocSysString(), startPageTempPath.AllocSysString());
+    if (!success)
+    {
+        // TODO: Handle error
+    }
+
+    return startPageTempPath;
+}
+
+CString Acrolinx_Sdk_Sidebar_Util::FileUtil::GetWebViewUserDataDirectory()
+{
+    WCHAR wcharPath[MAX_PATH];
+    CString webViewWorkingDir = _T("");
+    if (GetTempPathW(MAX_PATH, wcharPath))
+    {
+        webViewWorkingDir = wcharPath;
+        webViewWorkingDir = webViewWorkingDir + _T("Acrolinx\\");
+        CreateDirectory(webViewWorkingDir, NULL);
+        webViewWorkingDir = webViewWorkingDir + _T("webview-") + DllUtil::GetAppName() + _T("\\");
+        CreateDirectory(webViewWorkingDir, NULL);
+    }
+    else
+    {
+        LOGE << "Creating working directory for webview failed.";
+    }
+    return webViewWorkingDir;
+}
+
+bool Acrolinx_Sdk_Sidebar_Util::FileUtil::UnzipToFolder(BSTR lpZipFile, BSTR lpFolder)
```

Comment:
return BOOLEAN values instead of 1 and 0.

The bool will be checked to determine success or failure upon which startpage URL will be constructed or not